.github/workflows/development-environment.yml

```diff
@@ -0,0 +1,97 @@
+
+jobs:
+  dev-environment:
+    name: Python set up
+    runs-on: ${{ matrix.os }}
+    timeout-minutes: 20
+    strategy:
+      matrix:
+        os: [ macos-11.0 ]
+
+    steps:
+      - uses: actions/checkout@v2
+
+      - name: Install prerequisites
+        if: ${{ runner.os == 'macOS' }}
+        # Depending on the host watchman will not install (read: intermittent issue),
+        # thus, we mark this as continue-on-error
+        continue-on-error: true
+        # Xcode CLI & brew are already installed, thus, no need to call xcode-select install
+        # XXX: Can the Brew set up be cached?
+        run: |
+          brew bundle
+
+      - name: Set environment variables
+        shell: bash
+        run: |
+          echo "PYENV_ROOT=$HOME/.pyenv" >> $GITHUB_ENV
+          echo "VOLTA_HOME=$HOME/.volta" >> $GITHUB_ENV
+          echo "PATH=$HOME/.pyenv/shims:$HOME/.volta/bin:$PATH" >> $GITHUB_ENV
+
+      - name: Install Python (via Pyenv)
+        id: python-version
+        shell: bash
+        # XXX: Locally, pyenv makes 3.6.10 be the selected version in use,
+        # however, in here it selects /usr/local/bin/python. I'm using `pyenv global` to work around it
+        # XXX: Can we cache the installed Python environments?
+        run: |
+          make setup-pyenv
+          pyenv global 3.6.10
+          echo "::set-output name=python-version::$(python -V  | sed s/Python\ //g)"
+
+      - name: Setup pip
+        uses: ./.github/actions/setup-pip
+        id: pip
+
+      - name: Cache
+        uses: actions/cache@v2
+        with:
+          path: |
+            ${{ steps.pip.outputs.pip-cache-dir }}
+            ~/.pyenv
+          key: ${{ runner.os }}-pip-py${{ steps.python-version.outputs.python-version }}-${{ hashFiles('**/requirements-*.txt') }}
+          restore-keys: |
+            ${{ runner.os }}-pip-py${{ steps.python-version.outputs.python-version }}
+
+      - name: Set up development environment (as per docs)
+        run: |
+          python -m pip install virtualenv
+          python -m virtualenv .venv
+          source .venv/bin/activate
+          make install-py-dev
+          make setup-git
+          curl https://get.volta.sh | bash
+
+      - name: Set up Sentry and others
+        # We cannot call make bootstrap directly since run-dependent-services requires some Docker magic.
+        # We have the magic in the getsentry/bootstrap-develop. We will handle this situation in the next pass of this.
+        # make init-config will *not* prompt about overwriting ~/.sentry/config.yml' since we're on a pristine state
+        # We cannot test the git hooks until the `sentry.lint.engine` gets installed
+        run: |
+          source .venv/bin/activate
+          make develop init-config
+          git config --global user.email "you@example.com"
+          git config --global user.name "Your Name"
+          touch foo.txt && git add foo.txt && git commit -m "Test git hooks"
```

Comment:
How do we know if git hooks ran successfully?

Reply:
if we're testing git hooks we probably want to include these two files in the paths to check to run:

![image](https://user-images.githubusercontent.com/79684/102803411-0b98ed80-436d-11eb-8a64-9613b9830f31.png)

Reply:
For now, I've decided to drop this.

The reason I added this is because I've encountered myself many times trying to commit code without fully setting up the venv (perhaps because I kept removing and recreating it to test various work) because it requires `pip install -e .` to execute. I know I can use `--no-verify`. I will think this through properly if there's a good enough reason to add it.